core/io/resource.cpp

```diff
@@ -56,32 +56,32 @@ void Resource::set_path(const String &p_path, bool p_take_over) {
 		p_take_over = false; // Can't take over an empty path
 	}
 
-	ResourceCache::lock.lock();
+	{
+		MutexLock locked(ResourceCache::resources_mutex);
```

Comment:
```suggestion
		MutexLock lock(ResourceCache::resources_mutex);
```
Same for the rest, looks better (used `locked` because `lock` was taken, but now not needed)